tests/unit/lib/SGSparseMatrix_unittest.cc

```diff
@@ -265,3 +265,28 @@ TEST(SGSparseMatrix, from_dense)
 		for (index_t vecIndex=0; vecIndex<numberOfVectors; ++vecIndex)
 			EXPECT_EQ(sparseMatrix(featIndex,vecIndex), denseMatrix(featIndex,vecIndex));
 }
+
+TEST(SGSparseMatrix, transposed_square_matrix)
+{
+	const float64_t sparse_level=0.1;
+	const index_t number_of_features=100;
+	const index_t number_of_vectors=100;
+	const index_t rand_seed=0;
+
+	SGSparseMatrix<float64_t> sparse_matrix(number_of_features, number_of_vectors);
+	GenerateMatrix<SGSparseMatrix<float64_t> >(sparse_level, number_of_features, number_of_vectors, rand_seed, &sparse_matrix);
+
+	SGSparseMatrix<float64_t> sparse_matrix_t=sparse_matrix.get_transposed();
+
+	
+	//check dimension
```

Comment:
If you have an ```mxn``` design matrix, ```m``` training samples and ```n``` features, it is checking for equality, the no. of features and no. of training samples in the original and transposed matrices. Btw, this is a square matrix, so ```m=n```, which begs the question do we want to do it twice ? I think we should, better you do a check for training samples and the no. of features separately.